### PR TITLE
Create branching form template and used it with Grant Form Page and M…

### PR DESCRIPTION
**Pull request by Mo-sissoko**
### ℹ️ Issue

Closes https://trello.com/c/cHayfL7J/12-create-branching-form-template

### 📝 Description

Briefly list the changes made to the code:

- Added template for branching forms as listed in the ticket
- Update medical form page to use branching to confirm whether the person filling out the application is a medical professional
- Update grant form page to use branching to confirm that the person filling out the grant form is a parent/legal guardian

### ✔️ Verification

What steps did you take to verify your changes work? These should be clear enough for someone to be able to clone the branch and follow the steps themselves. 

Provide screenshots of any new components, styling changes, or pages. 

image

image

image

image

### 🏕️ (Optional) Future Work / Notes

Did you notice anything ugly during the course of this ticket? Any bugs, design challenges, or unexpected behavior? Write it down so we can clean it up in a future ticket!

### CODE REVIEW

File: apps/frontend/src/components/form/MedicalForm.tsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /* eslint-disable react/jsx-props-no-spreading */
```
Comment (nate-kirschner) [inline apps/frontend/src/components/form/MedicalForm.tsx]: Did anything change in this file?

File: apps/frontend/src/components/form/MedicalForm.tsx [hunk 2]
```diff
@@ -1,5 +1,5 @@
...
-import React from 'react';
+import React, { useState } from 'react';
```
Comment (nate-kirschner) [inline apps/frontend/src/components/form/MedicalForm.tsx]: Did anything change in this file?

File: apps/frontend/src/enums/Branches.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+export enum Branches {
```
Comment (nate-kirschner) [inline apps/frontend/src/enums/Branches.ts]: For something like this where there's no 'display' value or a reason to have two versions of each option, I'd prefer to just use a union type like 
`type Branches = 'CONFIRMATION' | 'FORM' | 'END';`
Not a necessary change though.

File: apps/frontend/src/pages/FormPage.tsx [hunk 4]
```diff
@@ -26,21 +65,47 @@ const FormPage: React.FC = () => {
...
+              endMessage={'Please find a medical professional to complete the form.'}
+              onConfirm={handleMedicalProfessional}
+            />
+          )}
 
-          <Center>
-            <Button mt={4} colorScheme="teal" onClick={form.submitForm}>
-              Submit
-            </Button>
+          <Center mt={4}>
```
Comment (nate-kirschner) [inline apps/frontend/src/pages/FormPage.tsx]: Maybe consider making these buttons a separate component. Not necessary though

File: apps/frontend/src/components/form/MedicalForm.tsx (hunk 1 above)
Comment (Mo-sissoko) [reply apps/frontend/src/components/form/MedicalForm.tsx]: Updated in most recent commit

File: apps/frontend/src/enums/Branches.ts (hunk 3 above)
Comment (Mo-sissoko) [reply apps/frontend/src/enums/Branches.ts]: Updated in most recent commit

File: apps/frontend/src/pages/FormPage.tsx (hunk 4 above)
Comment (Mo-sissoko) [reply apps/frontend/src/pages/FormPage.tsx]: Updated in most recent commit

**Comment by nate-kirschner** [comment]
This looks really good. Instead of adding this directly into the FormPage, lets leave the FormPage component alone for now, and create a BranchingFormPage that implements what you have in FormPage. The BranchingFormPage just won't be used yet, I think we're not quite ready to have this as the front page of the form yet.

**Review by nate-kirschner — APPROVED** [review]

**Pull request merged**